http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java

```diff
@@ -288,6 +297,38 @@ private static String getAccessTokenHash(String accessTokenString, String jwsAlg
 
     }
 
+
+    private static class NonceValidator implements ErrorCodeValidator {
+        private final String stateCookieValue;
+
+        public NonceValidator(final String stateCookieValue) {
+            this.stateCookieValue = stateCookieValue;
+        }
+
+        public ErrorCodeValidator.Error validate(JwtContext jwtContext) throws MalformedClaimException {
+            JwtClaims idJwtClaims = jwtContext.getJwtClaims();
+            IDToken idToken = new IDToken(idJwtClaims);
+
+            String[] cookieArr = stateCookieValue.split(";");
```

Comment:
As mentioned above, this shouldn't be needed.

Reply:
done